File: app/src/main/java/com/dantsu/thermalprinter/ItemAdapter.java
package com.dantsu.thermalprinter;


import android.annotation.SuppressLint;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.dantsu.thermalprinter.data.Comment;
import com.dantsu.thermalprinter.data.CommentData;
import com.squareup.picasso.Picasso;

import java.io.IOException;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.List;

import de.hdodenhof.circleimageview.CircleImageView;

public class ItemAdapter extends RecyclerView.Adapter<ItemAdapter.ItemViewHolder> {

    private List<Comment> items;

    private OnItemClickListener listener;


    public interface OnItemClickListener {
        void onItemClick(int position);
    }

    public void setOnItemClickListener(OnItemClickListener listener){
        this.listener = listener;
    }



    public ItemAdapter(List<Comment> items) {
        this.items = items;
    }

    @NonNull
    @Override
    public ItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_list, parent, false);
        return new ItemViewHolder(view, listener);
    }

    @Override
    public void onBindViewHolder(@NonNull ItemViewHolder holder, int position) {
        Comment currentItem = items.get(holder.getBindingAdapterPosition());
        holder.titleTextView.setText(currentItem.nickname);
        holder.descriptionTextView.setText(currentItem.comment);


        holder.countOrder.setText(String.valueOf(currentItem.countOrder));
        Picasso.get()
                .load(currentItem.profilePictureUrl)
                .fit()
                .into(holder.avatar);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public static class ItemViewHolder extends RecyclerView.ViewHolder {
        public TextView titleTextView;
        public TextView descriptionTextView;

        public TextView countOrder;
        public CircleImageView avatar;

        public Button printButton;

        public ItemViewHolder(@NonNull View itemView, final OnItemClickListener listener) {
            super(itemView);
            titleTextView = itemView.findViewById(R.id.textViewTitle);
            descriptionTextView = itemView.findViewById(R.id.textViewDescription);
            countOrder = itemView.findViewById(R.id.tvCountOrder);
            avatar = itemView.findViewById(R.id.avatarImageView);
            printButton = itemView.findViewById(R.id.trailingButton);
            printButton.setOnClickListener(view -> {
                if(listener!=null){
                    int position = getBindingAdapterPosition();
                    if (position != RecyclerView.NO_POSITION) {
                        listener.onItemClick(position);
                    }
                }
            });
        }
    }
}

File: app/src/main/java/com/dantsu/thermalprinter/ItemAdapter.kt
package com.dantsu.thermalprinter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.dantsu.thermalprinter.ItemAdapter.ItemViewHolder
import com.dantsu.thermalprinter.data.Comment
import com.squareup.picasso.Picasso
import de.hdodenhof.circleimageview.CircleImageView

class ItemAdapter(private val items: List<Comment>) : RecyclerView.Adapter<ItemViewHolder>() {
    private var listener: OnItemClickListener? = null

    interface OnItemClickListener {
        fun onItemClick(position: Int)
    }

    fun setOnItemClickListener(listener: OnItemClickListener?) {
        this.listener = listener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_list, parent, false)
        return ItemViewHolder(view, listener)
    }

    override fun onBindViewHolder(holder: ItemViewHolder, position: Int) {
        val currentItem = items[holder.bindingAdapterPosition]
        holder.titleTextView.text = currentItem.nickname
        holder.descriptionTextView.text = currentItem.comment
        Picasso.get()
            .load(currentItem.profilePictureUrl)
            .fit()
            .into(holder.avatar)
    }

    override fun getItemCount(): Int {
        return items.size
    }

    class ItemViewHolder(itemView: View, listener: OnItemClickListener?) : ViewHolder(itemView) {
        var titleTextView: TextView
        var descriptionTextView: TextView
        var avatar: CircleImageView
        var printButton: Button

        init {
            titleTextView = itemView.findViewById(R.id.textViewTitle)
            descriptionTextView = itemView.findViewById(R.id.textViewDescription)
            avatar = itemView.findViewById(R.id.avatarImageView)
            printButton = itemView.findViewById(R.id.trailingButton)
            printButton.setOnClickListener {
                if (listener != null) {
                    val position = bindingAdapterPosition
                    if (position != RecyclerView.NO_POSITION) {
                        listener.onItemClick(position)
                    }
                }
            }
        }
    }
}